handler/src/main/java/io/netty/handler/ssl/util/BouncyCastlePemReader.java

```diff
@@ -0,0 +1,169 @@
+        try {
+            return getPrivateKey(getParser(keyInputStream), keyPassword);
+        } catch (Exception e) {
+            logger.debug("Unable to extract private key due to " + e);
+            return null;
+        }
+    }
+
+    public static PrivateKey getPrivateKey(File keyFile, String keyPassword) {
+        try {
+            return getPrivateKey(getParser(keyFile), keyPassword);
+        } catch (Exception e) {
+            logger.debug("Unable to extract private key due to " + e);
+            return null;
+        }
+    }
+
+    private static PrivateKey getPrivateKey(PEMParser pemParser, String keyPassword) throws PEMException, IOException,
+            PKCSException, OperatorCreationException {
+
+        try {
+            Object object = pemParser.readObject();
+            logger.debug("Parsed PEM object of type " + object.getClass().getName() + " and assume key is "
+                    + (keyPassword == null ? "not " : "") + "encrypted");
+            JcaPEMKeyConverter converter = new JcaPEMKeyConverter().setProvider("BC");
+            PrivateKey pk = null;
+
+            if (keyPassword == null) {
+                // assume private key is not encrypted
+                if (object instanceof PrivateKeyInfo) {
+                    pk = converter.getPrivateKey((PrivateKeyInfo) object);
+                } else if (object instanceof PEMKeyPair) {
+                    pk = converter.getKeyPair((PEMKeyPair) object).getPrivate();
+                } else {
+                    logger.debug("Unable to handle PEM object of type "
+                                   + object.getClass() + " as a non encrypted key");
+                }
+            } else {
+                // assume private key is encrypted
+                if (object instanceof PEMEncryptedKeyPair) {
+                    PEMDecryptorProvider decProv = new JcePEMDecryptorProviderBuilder()
+                            .build(keyPassword.toCharArray());
+                    pk = converter.getKeyPair(((PEMEncryptedKeyPair) object).decryptKeyPair(decProv)).getPrivate();
+                } else if (object instanceof PKCS8EncryptedPrivateKeyInfo) {
+                    InputDecryptorProvider pkcs8InputDecryptorProvider = new JceOpenSSLPKCS8DecryptorProviderBuilder()
+                            .build(keyPassword.toCharArray());
+                    pk = converter.getPrivateKey(((PKCS8EncryptedPrivateKeyInfo) object)
+                            .decryptPrivateKeyInfo(pkcs8InputDecryptorProvider));
+                } else {
+                    logger.debug("Unable to handle PEM object of type " + object.getClass() + " as a encrypted key");
+                }
+            }
+
+            if (pk == null) {
+                logger.debug("No key found");
+            }
+
+            return pk;
+        } finally {
+            if (pemParser != null) {
+                try {
+                    pemParser.close();
+                } catch (Exception e) {
```

Comment:
e -> ignored

Reply:
what else can we do here instead of ignoring?

Reply:
@floragunncom I only asked to rename `e` to `ignored` ;)

Reply:
You may also want to log something in to debug level

Reply:
changed to "ignored"